### ISSUE TITLE
Expressing Action Prerequisite or "after"

### ISSUE THREAD

**Issue by adomokos** [adomokos/light-service#97]
Usually, the order of actions are dependent on previous actions, but I caught myself rearranging orders where it makes sense. However, sometimes the action has to be invoked after another one.

Would it make sense to introduce an `after` macro that would express this dependency?

So the action would look like this:
```ruby
class SomeAction
  extend LightService::Action

  after SomeOtherAction
  expects :user
  promises :client

  executed do |ctx|
    ...
  end
end
```

And when somebody places this action into a workflow where the `SomeOtherAction` was not invoked prior to `SomeAction#execute` call, it will throw an error.

I am hesitant to add this logic as it could be abused, but maybe, somebody else bumped into this before.

Let me know what you think!

**Comment by Jared-Prime** [comment]
The formalization of this pattern starts to make my brain hurt ( cf. [ordered monoids](http://www.math.uiuc.edu/Macaulay2/doc/Macaulay2-1.9/share/doc/Macaulay2/Macaulay2Doc/html/___Ordered__Monoid.html) ). My confusion revolves around the question: once a programmer imposes an ordering on a single action, should she make explicit that ordering on all the actions under that organizer? I would say "yes"; the trouble, then, would be that the action where ordering is expressed as suggested would then require an ordering in any organizer that includes that action. The way to resolve that would simply be to express the ordering under the organizer, and leave the action agnostic to its caller.

In short, it would seem to me that an action should have no knowledge of its fellow actions; an action should only have knowledge about its context.

I would like to see more examples where the proposed change is useful, tho

**Comment by adomokos** [comment]
This would add more restriction to an action with limiting the options for reuse. It was an idea, thanks for taking the time and thinking about it @Jared-Prime. I am closing it.
